### PR TITLE
n-jeft transistor

### PR DESCRIPTION
**Pull request by aman1376**
ref: #44 

Screenshot 2024-10-06 at 2 39 10 AM
Screenshot 2024-10-06 at 2 39 33 AM

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
should be "n-jfet"

also make sure to run `bun test` to generate the snapshot

File: symbols/n_jeft_transistor_horz.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  primitives: [
+    ...Object.values(paths),
+    ...Object.values(circles),
+    { ...texts.top1, anchor: "middle_center" },
+    { ...texts.bottom1, anchor: "middle_center", x:-0.15},
+  ] as any,
+  ports: [
+    { ...refblocks.top1, labels: ["1"] }, // TODO add more "standard" labels
+    { ...refblocks.bottom1, labels: ["2"] }, // TODO add more "standard" labels
+    { ...refblocks.left1, labels: ["3"] }, // TODO add more "standard" labels
```
Comment (seveibar) [inline symbols/n_jeft_transistor_horz.ts:16]: can you add "gate", "source" and "drain" here

**Review by seveibar — CHANGES_REQUESTED** [review]

File: symbols/n_jeft_transistor_horz.ts (hunk 1 above)
Comment (aman1376) [reply symbols/n_jeft_transistor_horz.ts]: done

File: symbols/n_jfet_transistor_horz.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  primitives: [
+    ...Object.values(paths),
+    ...Object.values(circles),
+    { ...texts.top1, anchor: "middle_center" },
+    { ...texts.bottom1, anchor: "middle_center", x: -0.15 },
+  ] as any,
+  ports: [
+    { ...refblocks.top1, labels: ["Drain"] }, // TODO add more "standard" labels
+    { ...refblocks.bottom1, labels: ["Source"] }, // TODO add more "standard" labels
+    { ...refblocks.left1, labels: ["Gate"] }, // TODO add more "standard" labels
```
Comment (seveibar) [inline symbols/n_jfet_transistor_horz.ts:16]:
```suggestion
    { ...refblocks.top1, labels: ["1", "drain"] }, // TODO add more "standard" labels
    { ...refblocks.bottom1, labels: ["2", "source"] }, // TODO add more "standard" labels
    { ...refblocks.left1, labels: ["3", "gate"] }, // TODO add more "standard" labels
```

Comment (seveibar) [reply symbols/n_jfet_transistor_horz.ts:16]: always lowercase, we also always want to keep the pin numbers

**Review by seveibar — APPROVED** [review]
see comment/suggestion, after that looks good to go

**Comment by aman1376** [comment]
bun test is failing should i push this again?

**Comment by Timer00** [comment]
> bun test is failing should i push this again?

You can run ``bun test --update-snapshots`` and push the update

**Comment by Timer00** [comment]
One more small change, didn't realize about this one before mb! 
The text anchor should be positioned in the opposite direction of where the text may grow. 
image
On the right side looks great, on the left side if that text is longer it would go over the symbol. 
So in this case, simply move the texts anchor (blue dot) to the right side.

**Pull request closed without merging**